OMCompiler/Compiler/BackEnd/BackendDAECreate.mo

```diff
@@ -223,14 +226,45 @@ algorithm
 
   //remove alias from extVarArray
   extVarsOut := BackendVariable.deleteVars(extAliasVars,extVars);
+  extVarsOut := removeExtAliasBinding(extVarsOut,repl);
 
   //replace in equations
   (oEqs,_) := BackendVarTransform.replaceEquations(oEqs,repl,NONE());
   (oInitEqs,_) := BackendVarTransform.replaceEquations(oInitEqs,repl,NONE());
+  (oRemEqs,_) := BackendVarTransform.replaceEquations(oRemEqs,repl,NONE());
+
   oEqs := listReverse(oEqs);
   oInitEqs := listReverse(oInitEqs);
+  oRemEqs := listReverse(oRemEqs);
 end getExternalObjectAlias;
 
+
+protected function removeExtAliasBinding "Removes the binding of an external variable if the binding cref ist already detected as an alias variable.
+author: Waurich TUD 2020-05"
+  input BackendDAE.Variables extVarsIn;
+  input BackendVarTransform.VariableReplacements repl;
+  output BackendDAE.Variables extVarsOut;
+protected
+  list<BackendDAE.Var> varLst,extVarLst;
+  DAE.ComponentRef cref;
+algorithm
+  varLst := BackendVariable.varList(extVarsIn);
+  extVarLst := {};
+  for var in varLst loop
+		var := matchcontinue(var)
```

Comment:
I do not understand why matchcontinue is used here, hasReplacement might fail.
Something I have missed?

Also check the indention it is currently not symetric

Reply:
Old coding style, i guess force of habit